gui/src/pages/NewProject.svelte

```diff
@@ -1,31 +1,52 @@
 <script lang="ts">
   import Layout from '../components/Layout.svelte';
+  import Spinner from '../components/Spinner.svelte';
   import CenterFormPanel from '../components/form/CenterFormPanel.svelte';
   import * as router from 'svelte-spa-router';
 
-  const starters = [
-    {
-      displayName: 'Empty project',
-      name: 'empty-project',
+  const api = {
+    kernel: {
+      describeTemplates: async () => [
+        {
+          name: 'empty-project',
+          displayName: 'Empty project',
+          url: 'https://github.com/deref/exo-starters/empty-project',
+        },
+        {
+          name: 'nextjs-prisma',
+          displayName: 'NextJS Prisma',
+          url: 'https://github.com/railwayapp/starters/tree/master/examples/nextjs-prisma',
+        },
+        {
+          name: 'http-nodejs',
+          displayName: 'HTTP Module',
+          url: 'https://github.com/railwayapp/starters/tree/master/examples/http-nodejs',
+        },
+        {
+          name: 'laravel',
+          displayName: 'Laravel Starter',
+          url: 'https://github.com/railwayapp/starters/tree/master/examples/laravel',
+        },
+      ],
     },
-    {
-      displayName: 'Next.js Prisma Postgres App',
-      name: 'next-prisma-postgres',
-    },
-  ];
+  };
 </script>
 
 <Layout>
   <CenterFormPanel title="New project" backRoute="/">
     <h1>New project</h1>
     <p>Select a starter for your new project:</p>
-    {#each starters as starter}
-      <button
-        on:click={() => {
-          router.push(`#/new-project/${starter.name}`);
-        }}>{starter.displayName}</button
-      >
-    {/each}
+    {#await api.kernel.describeTemplates()}
```

Comment:
I'm not actually sure how this works in Svelte. My intuition is that this will cause an API request every time the component is rendered, but maybe it only causes an API request when `api` changes?

Reply:
Yes that's my intuition as well — though I can't immediately find a conclusive answer. My approach to this was to put the promise in the initial script tag which is run just on creation of the component. (https://svelte.dev/docs#script)

Reply:
I don't understand why this would be a problem in this case - can somebody please elaborate the source of concern?

Reply:
The fear here was that on every render you'd get a new request. I added the following to the top level script tag to test it:
```
  // added this to top level script tag
  let v = 1;
  setInterval(() => v++, 1000);
```

I also added a `<p>{v}</p>` to the body.

But I only got one request. So I think it's all fine.

Reply:
Okay, cool. I still don't really understand why that fear is a concern - seems like there wouldn't be many renders in the first place, nor does the function seem expensive. Concern makes sense if these assumptions are not true though.

Reply:
I mean I don't think it would ever have been a massive problem for this page as it is. But as a pattern it would be bad if every time anything changes on the page we refetch all data.